gcloud-java-dns/src/test/java/com/google/gcloud/dns/it/ITDnsTest.java

```diff
@@ -587,6 +586,75 @@ public void testCreateChange() {
   }
 
   @Test
+  public void testInvalidChangeRequest() {
+    Zone zone = DNS.create(ZONE1);
+    DnsRecord validA = DnsRecord.builder("subdomain." + zone.dnsName(), DnsRecord.Type.A)
+        .records(ImmutableList.of("0.255.1.5"))
+        .build();
+    try {
+      ChangeRequest validChange = ChangeRequest.builder()
+          .add(validA)
+          .build();
+      zone.applyChangeRequest(validChange);
+      try {
+        zone.applyChangeRequest(validChange);
+        fail();
+      } catch (DnsException ex) {
+        // expected
+        assertFalse(ex.retryable());
+        assertEquals(409, ex.code());
+      }
+      // delete with field mismatch
+      DnsRecord mismatch = validA.toBuilder().ttl(20, TimeUnit.SECONDS).build();
+      ChangeRequest delete = ChangeRequest.builder().delete(mismatch).build();
+      try {
+        zone.applyChangeRequest(delete);
+        fail();
+      } catch (DnsException ex) {
+        // expected
+        assertEquals(412, ex.code());
+        assertFalse(ex.retryable());
+      }
+      // delete and add SOA
+      Iterator<DnsRecord> recordIterator = zone.listDnsRecords().iterateAll();
+      LinkedList<DnsRecord> deletions = new LinkedList<>();
+      LinkedList<DnsRecord> additions = new LinkedList<>();
+      while (recordIterator.hasNext()) {
+        DnsRecord record = recordIterator.next();
+        if (record.type() == DnsRecord.Type.SOA) {
+          deletions.add(record);
+          DnsRecord copy = record.toBuilder().name("x." + record.name()).build();
+          additions.add(copy);
+          break;
+        }
+      }
+      delete = delete.toBuilder().deletions(deletions).build();
+      ChangeRequest addition = ChangeRequest.builder().additions(additions).build();
```

Comment:
Nit: could make the variable names for the ChangeRequest consistent (deletion and addition or delete and add)